doc/design/web.md

```diff
@@ -0,0 +1,141 @@
+    - Download SSL keys
+- Jupiter Notebook
+    - Private Jupiter Notebook environment to run Python scripts
+    - Private workspace
+    - Submit job from Jupiter Notebook
+- Job Dashboard
+    - Job history and current running jobs
+    - Performance Monitoring
+    - Quota Monitoring
+- Datasets
+    - Public Dataset viewing
+    - Upload/Download private datasets
+    - Share datasets
+- Models
+    - Upload/Download models file
+    - Share/Publish Models
+- Paddle Board
+    - Training metrics visualization
+        - cost
+        - evaluator
+        - user-defined metrics
+- Serving
+    - Submit serving instances
+    - Deactivate serving
+    - Serving performance monitoring
+
+## Account Management
+
+I'll skip this section because it is a design that almost every website need.
+
+## Jupiter Notebook
+
+Start a ReplicaSet using image `docker.paddlepaddle.org/book` in kubernetes cluster and add an ingress endpoint when user first enters the notebook page.
+
+<img src="pictures/notebook.png" width="500px" align="center">
+
+Users can write program in python in the web page and save their programs, which will be saved at cloud storage. Users also can run script like below to submit a cluster training job:
+
+```python
+sess = paddle.framework.remote_session(
+    topology=block,
+    reader=reader,
+    jobname="cluster_job",
+    gpu=12
+)
+sess.run()
+```
+
+After this, there will be a job description and perfomance monitoring pages able to view at "Job Dashboard"
+
+## Job Dashboard
+
+### Job history and current running jobs
+
+A web page containing a table to list jobs satisfing user's filters. User can only list jobs that was submited by themselves.
+
+| jobname | start time | age      | success | fails | actions |
+| ------- | ---------- | -------- | ------- | ----- | ------- |
+| test1   | 2017-01-01 | 17m      |    0    |   0   | stop/log/perf |
+
+Users can filter the list using:
+
+- status:  running/stoped/failed
+- time:    job start time
+- jobname: search by jobname
+
+Viewing job logs:
+
+Click the "log" button on the right of the job will pop up a console frame at the bottom of the page showing the tail of the job log, here shows the first pod's log. On the left side of pop up console frame, there should be a vertical list containing the list of pods, when click one of the pod, the console will show the log of it.
+
+### Performance Monitoring
+
+A web page containing graphs monitoring job's resource usages according to time change:
+
+- CPU usage
+- GPU usage
+- memory usage
+- network bandwidth
+- disk I/O
+
+### Quota Monitoring
+
+A web page displaying total quota and quota currently in use.
+
+Also display total CPU time, GPU time in latest 1day, 1week and 1month.
+
+## Datasets and Models
+
+Datasets and Models are quite the same, both like a simple file management and sharing service.
+
+- file listing and viewing page
+- Upload/Download page
+- file sharing page
+
+## Paddle Board
```

Comment:
I didn't expect `draw_board` functions calls in user programs. I am not sure how configurable the TensorBoard is, but in my mind, PaddleBoard just needs to be able to present outputs from Evaluator operators aggregated/accumulated over minibatches.

Reply:
If not inserting function calls in user programs, we need to automatically find out which variable represents the cost and evaluator operator by default and draw the variable value to the web page. I'm not sure how to do that for now. 

[Here](https://www.tensorflow.org/get_started/summaries_and_tensorboard) is a short example of how TensorBoard config the metrics, using `tf.summary`. User explicitly specify values to output for drawing.